src/AbstractActionHandler.ts

```diff
@@ -104,15 +104,15 @@ export abstract class AbstractActionHandler implements ActionHandler {
     }
     // Only check if this is the block we need if it's not the first block
     if (!isEarliestBlock) {
```

Comment:
Probably cleaner if the check is on this line: `if (!isEarliestBlock && this.validateBlocks)`

Reply:
Done!